scrc/preprocessors/extractors/section_splitter.py

```diff
@@ -61,6 +65,40 @@ def read_column(self, engine: Engine, spider: str, name: str, lang: str) -> pd.D
         query = f"SELECT count({name}) FROM {lang} WHERE {self.get_database_selection_string(spider, lang)} AND {name} <> ''"
         return pd.read_sql(query, engine.connect())['count'][0]
 
+    """log_coverage for users without database write privileges, parses results from json logs
+       allows to print coverage half way through with the coverage for the parsed chunks still being valid"""
+    def log_coverage_from_json(self, engine: Engine, spider: str, lang: str, batch_info: dict) -> None:
+        df = pd.DataFrame()
+        path = Path.joinpath(self.output_dir, os.getlogin(), spider, lang, batch_info['uuid'])
+        # retrieve all chunks
+        for chunk in range(batch_info['chunknumber']):
+            df_chunk = pd.read_json(str(path / f"{chunk}.json"))
+            df = df.append(df_chunk)
```

Comment:
This can be a bit problematic in large spiders because it might overload the RAM. This is the reason for using chunks in the first place. Is there a way of not loading all the chunks at the same time?
Sorry if I misunderstood something here.

Reply:
Very good idea. It should not be a problem this way, as it runs when either the spider just started and we want to get coverage for the first few batches when debugging or when the spider has finished and memory should already be freed up again. But why waste memory if we can do it much more efficient.

I can just summarize for each file.

Reply:
Should be fine now, only concern is that if the log is printed at the very end and the garbage collector is slow on the overwritten df_chunk dataframe, it might use a lot of memory. But I think we can rely on the garbage collector to perform well and if it should become a problem in the future we can still fix it.

Reply:
Ok, great. Thanks.